### PR TITLE
Production demoshop

### PR DESCRIPTION
**Pull request by boris-brtan**
| Q             | A
| ------------- | ---
|Description, reason for the PR| we want to have running demoshop on production server with natively installed data storages and also we would like to have modifiable administration url.
|New feature| Yes 
|BC breaks| No 
|Fixes issues| 
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: docker/php-fpm/Dockerfile-production [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+USER www-data
+
+COPY docker/php-fpm/php-ini-overrides.ini /usr/local/etc/php/php.ini
+COPY --chown=www-data:www-data / /var/www/html
+
+RUN touch /var/www/html/PRODUCTION
+
+# allow configuring the GitHub OAuth token
+ARG github_oauth_token
+RUN composer config -g github-oauth.github.com $github_oauth_token
```
Comment (MattCzerner) [inline docker/php-fpm/Dockerfile-production]: Do we still need it? If we know that it is not necessary anymore, then why to use it?

**Review by MattCzerner — CHANGES_REQUESTED** [review]
I guess the code is fine. I wonder about changelog, is it generated? Or not yet? 

I would like to have the script on jenkins in the repository so it is versioned, of course that means to make script more parametirized. I added comments right into script on jenkins.

Is a script on jenkins from some previous US? I dont know if I should comment it or not.

File: docker/php-fpm/Dockerfile-production (hunk 1 above)
Comment (boris-brtan) [reply docker/php-fpm/Dockerfile-production]: yes, demoshop is now in version beta1 and composer.json has got repositories section.
and we have just created issue for this. Maybe we can set higher priority for that issue

**Review by MattCzerner — APPROVED** [review]

**Comment by MattCzerner** [comment]
After internal discussion We decided not to create script file, because it is already in monorepo documentation.

Tests are okay by me.

**Comment by boris-brtan** [comment]
please @MattCzerner can you do another CR for fixup and https://github.com/shopsys/shopsys/pull/530 ? THX.
there was one problem that if admin is not logged and go to administration via second domain, user is redirected onto %admin_url% instead of the value of the admin_url parameter.
Example:
```
parameters: admin_url: admin1234
domain1: shop.com
domain2: 2.shop.com
```
redirection from `http://2.shop.com/admin1234` to `http://shop.com/%admin_url%` is wrong and should be `http://shop.com/admin1234` and after login it should be redirected back to `http://2.shop.com/admin1234/dashboard`

File: docker/php-fpm/Dockerfile-production [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+# overwrite the original entry-point from the PHP Docker image with our own
+COPY docker/php-fpm/docker-php-entrypoint /usr/local/bin/
+
+# allow overwriting UID and GID o the user "www-data" to help solve issues with permissions in mounted volumes
+RUN chown www-data:www-data /var/www/html
+USER www-data
+
+COPY docker/php-fpm/php-ini-overrides.ini /usr/local/etc/php/php.ini
+COPY --chown=www-data:www-data / /var/www/html
```
Comment (MattCzerner) [inline docker/php-fpm/Dockerfile-production]: The parameters and domains_urls creation was deleted here, but it is not created even on CI? Where it its time? Is it tested?

File: docker/php-fpm/Dockerfile-production [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+# see https://github.com/docker-library/php/issues/240#issuecomment-305038173
+RUN apk add --update --no-cache --repository http://dl-cdn.alpinelinux.org/alpine/edge/testing gnu-libiconv
+ENV LD_PRELOAD /usr/lib/preloadable_libiconv.so php
+ENV LC_ALL=en_US.utf8 LANG=en_US.utf8 LANGUAGE=en_US.utf8
+
+# overwrite the original entry-point from the PHP Docker image with our own
+COPY docker/php-fpm/docker-php-entrypoint /usr/local/bin/
+
+# allow overwriting UID and GID o the user "www-data" to help solve issues with permissions in mounted volumes
+RUN chown www-data:www-data /var/www/html
```
Comment (MattCzerner) [inline docker/php-fpm/Dockerfile-production]: This is duplicated with line 65

**Comment by MattCzerner** [comment]
I wonder if demoshop should have multistage build too.. What do you think? I would create a US for that now, since we are trying to finish this

**Review by MattCzerner — CHANGES_REQUESTED** [review]

File: docker/php-fpm/Dockerfile-production (hunk 3 above)
Comment (boris-brtan) [reply docker/php-fpm/Dockerfile-production]: COPY --chown=www-data:www-data / /var/www/html  will copy content but will not set owner of /var/www/html to www-data user

File: docker/php-fpm/Dockerfile-production (hunk 2 above)
Comment (boris-brtan) [reply docker/php-fpm/Dockerfile-production]: parameters are created during composer install and and domains_urls are created during production build process https://github.com/shopsys/shopsys/blob/master/docs/installation/installation-using-docker-on-production-server.md#docker-image-building .

**Comment by boris-brtan** [comment]
it will have multistage build after another demoshop upgrade but does this make sense to point this version of docker for production onto the correct version of the documentation for installation on production server ?

File: docker/php-fpm/Dockerfile-production (hunk 3 above)
Comment (MattCzerner) [reply docker/php-fpm/Dockerfile-production]: Why do you use it then? Isnt it better to first copy all the files into directory and then chown it? Still duplicate in my mind

**Review by MattCzerner — APPROVED** [review]

**Pull request merged**